CSS mask-origin property (#39366)
mdn/content: files/en-us/web/css/mask-origin/index.md

mask-origin: unset;
```

### Values

The `mask-origin` property is a comma-separated list of `<coord-box>` keyword values, including:

- `content-box`
  - : The position is relative to the content box.

> - `padding-box`

Author: updated this with links

> - `padding-box`

Reviewer: that works perfectly

- `fill-box`
  - : The position is relative to the object bounding box.
- `stroke-box`
  - : The position is relative to the stroke bounding box.
- `view-box`
  - : Uses the nearest SVG viewport as reference box. If a {{svgattr("viewBox")}} attribute is specified for the element creating the SVG viewport, the reference box is positioned at the origin of the coordinate system established by the `viewBox` attribute and the dimension of the reference box is set to the width and height values of the `viewBox` attribute.

There are three non-standard values that are shortcuts for standard `<coord-box>` values: `content` is an alias for `content-box`, `padding` is an alias for `padding-box`, and `border` is an alias for `border-box`.

## Description

The `mask-origin` property is very similar to the {{cssxref("background-origin")}} property, but it has a different set of values and a different initial value. The initial value depends on the if there is an associated CSS layout box; if yes, the default value is `border-box`. In comparison, the default for `background-origin` is `padding-box`.

> For SVG elements without an associated CSS layout box, the values `content-box`, `padding-box` and `border-box` compute to `fill-box`; so the default `border-box` computes to `fill-box`, in which case the position is relative to the object bounding box. For HTML elements, if a SVG-related value of `fill-box`, `stroke-box`, or `view-box` is set, the value computes to the default `border-box`.

Reviewer: ```suggestion
For SVG elements without an associated CSS layout box, the values `content-box`, `padding-box`, and `border-box` (the default value) compute to `fill-box`, meaning the position is relative to the object bounding box. For HTML elements, if a SVG-related value of `fill-box`, `stroke-box`, or `view-box` is set, the value is computed to `border-box`.
```

## Examples

### Comparing content, padding, and border

This example demonstrates basic usage while comparing three values of the `mask-origin` property.

#### HTML

We include four {{htmlelement("section")}} elements, each containing one {{htmlelement("div")}} element.

```html
<section class="content">
  <div></div>
</section>
<section class="padding">
  <div></div>
</section>
<section class="border">
  <div></div>
</section>
<section class="comparison">
  <div></div>
</section>
```

#### CSS

> We apply {{cssxref("border")}}, {{cssxref("padding")}}, and {{cssxref("margin")}} to every `<div>`. These create the reference points for the mask image origin points. The `border` shorthand includes a {{cssxref("border-color")}}. We also include a {{cssxref("background-color")}}. These provide a green background and a blue border to mask. Lastly, all our `<div>` elements are provided with a {{cssxref("mask-image")}}.

Reviewer: ```suggestion
We apply {{cssxref("border")}}, {{cssxref("padding")}}, and {{cssxref("margin")}} to every `<div>`. These create the reference points for the mask image origin. The `border` shorthand includes a {{cssxref("border-color")}}. We also include a {{cssxref("background-color")}}. These provide a green background and a blue border to mask. Finally, all our `<div>` elements are provided with a {{cssxref("mask-image")}}.
```
